### PR TITLE
Support gRPC client and transmit native log

### PR DESCRIPTION
**Pull request by mrproliu**
* Support the gRPC client
* Support transmit native log to gRPC server
* Fix missing `insecure_skip_verify` field in demo config

### CODE REVIEW

File: configs/satellite_config.yaml [hunk 1]
```diff
@@ -49,6 +49,25 @@ sharing:
...
+      # The file path of client.key. The config only works when opening the TLS switch.
+      client_key_path: ${SATELLITE_GRPC_CLIENT_KEY_PATH:"client.key"}
+      # InsecureSkipVerify controls whether a client verifies the server's certificate chain and host name.
+      insecure_skip_verify: ${SATELLITE_GRPC_INSECURE_SKIP_VERIFY:false}
+      # The file path oca.pem. The config only works when opening the TLS switch.
+      ca_pem_path: ${SATELLITE_grpc_CA_PEM_PATH:"ca.pem"}
+      # How frequently to check the connection
+      check_period: ${SATELLITE_GRPC_CHECK_PERIOD:5}
+      # The auth value when send request
+      authentication: ${SATELLITE_GRPC_AUTHENTICATION:""}
```
Comment (wu-sheng) [inline configs/satellite_config.yaml:70]: Should we begin to change default pipeline? Use grpc client seems  better way?

**Review by wu-sheng — APPROVED** [review]
cc @gxthrj @EvanLjp

File: configs/satellite_config.yaml (hunk 1 above)
Comment (EvanLjp) [reply configs/satellite_config.yaml:70]: no problem

Comment (mrproliu) [reply configs/satellite_config.yaml:70]: Should I remove the Kafka client config or keep it?

Comment (wu-sheng) [reply configs/satellite_config.yaml:70]: Remove please, we have document about kafka client already, right?
And with you are making things completed, let's add e2e here.

File: internal/satellite/config/loader_test.go [hunk 2]
```diff
@@ -75,82 +75,103 @@ func params() *SatelliteConfig {
...
 		},
-		Sharing: &SharingConfig{
-			SharingCommonConfig: &config.CommonFields{
-				PipeName: "sharing",
+		Sharing: sharding(),
+		Pipes:   pipes(),
+	}
+}
+
+func sharding() *SharingConfig {
```
Comment (kezhenxu94) [inline internal/satellite/config/loader_test.go:83]: This is a typo, sharding -> sharing

Comment (mrproliu) [reply internal/satellite/config/loader_test.go]: Thx.

File: configs/satellite_config.yaml (hunk 1 above)
Comment (mrproliu) [reply configs/satellite_config.yaml:55]: Sure.

**Review by wu-sheng — APPROVED** [review]

**Review by kezhenxu94 — APPROVED** [review]
LGTM

**Pull request merged**